content/blog/2023/2023-10-25-spring-java-config.md

```diff
@@ -0,0 +1,1176 @@
+
+When you use Jasypt with Spring Boot, you can annotate your configuration classes with `@EnableEncryptableProperties`.
+
+Here's how you can use `@EnableEncryptableProperties` in a Spring Boot application:
+
+Example: Using @EnableEncryptableProperties with Jasypt in Spring Boot:
+
+**a. Add Jasypt Dependency:**
+
+Make sure you have the Jasypt Spring Boot starter dependency in your project.
+
+**Maven:**
+
+```xml
+<dependency>
+    <groupId>com.github.ulisesbocchio</groupId>
+    <artifactId>jasypt-spring-boot-starter</artifactId>
+</dependency>
+```
+
+**Gradle:**
+
+```gradle
+implementation 'com.github.ulisesbocchio:jasypt-spring-boot-starter:3.0.5' // Use the latest version
+```
+
+**b. Annotate Configuration Class with @EnableEncryptableProperties:**
+
+```java
+@Configuration
+@EnableEncryptableProperties
+@PropertySource(name="EncryptedProperties", value = "classpath:encrypted.properties")
+public class DatabaseConfiguration {
+    // Configuration beans and other application configurations
+}
+```
+
+In this example, the `@EnableEncryptableProperties` annotation enables property encryption for properties defined in .properties file.
+
+**c. Encrypt Properties in application.properties or application.yml:**
+
+```properties
+# application.properties
+myapp.password=ENC(encryptedpassword)
+```
+
+In your `application.properties` or `application.yml`, you can use the `ENC()` prefix to indicate that the property is encrypted. Jasypt will automatically decrypt the property during the application startup using the provided encryption key.
+
+Now when you do `environment.getProperty("secret.property")` or use `@Value("${secret.property}")` what you get is the decrypted version of that property.
+
+By using `@EnableEncryptableProperties`, you can secure sensitive properties in your Spring Boot application without having to manually decrypt them. This integration simplifies the process of encrypting and decrypting sensitive information, ensuring the security of your application's configuration data.
+
+**8. Conclusion: Empowering Configurability in Spring Applications:**
+Spring’s PropertySource provides a robust foundation for managing configuration data. By leveraging `PropertySource` and its integrations, developers can ensure that their applications are adaptable, secure, and maintainable. The ability to externalize properties and dynamically adjust configurations not only simplifies the development process but also lays the groundwork for resilient, flexible, and scalable Spring applications.
+
+## Seamless External Configuration
```

Comment:
As mentioned above, this isn't really what Spring's "Java Config" is about ... it's about "configurartion parameters for your application", not about "configuring your application context"

Reply:
Removed section on AnnotationConfigApplicationContext